src/main/java/io/dropwizard/kafka/KafkaTopicsFactory.java

```diff
@@ -0,0 +1,61 @@
+package io.dropwizard.kafka;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.dropwizard.jackson.Discoverable;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.hibernate.validator.constraints.NotEmpty;
+
+import javax.validation.constraints.Min;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class KafkaTopicsFactory extends ArrayList<KafkaTopicsFactory.KafkaTopic> implements Discoverable {
+
+    static class KafkaTopic {
+        @NotEmpty
+        @JsonProperty
+        private String name;
+
+        @Min(1)
+        @JsonProperty
+        private int partitions;
+
+        @Min(1)
+        @JsonProperty
+        private short replicationFactor;
+
+        @JsonProperty
+        Map<String, String> configs = null;
```

Comment:
Instead of `null`, maybe default it to `Collections.emptyMap()`?

Reply:
It's null in the Kafka source code. I think it should be okay.

Reply:
That comment was more in the interest of following the conventions I usually follow for these libraries. 

I tend to prefer default null objects over optionals (with null checks in the code), but empty collections by default. 

Not super strict on that, and I'm willing to let it slide, but it will differ from the other stuff in this library.